pkg/kubelet/kubelet.go

```diff
@@ -1776,6 +1777,51 @@ func (kl *Kubelet) recordNodeOnlineEvent() {
 	kl.recorder.Eventf(kl.nodeRef, "online", "Node %s is now online", kl.hostname)
 }
 
+// Returns the most recent sys oom event from cadvisor.
+// Returns nil if no sys oom events were observed.
+func (kl *Kubelet) getRecentSysOOMEvent() (*cadvisorApi.OomEventData, util.Time, error) {
+	timestamp := util.Time{}
+	request := &cadvisorEvents.Request{
+		EventType: map[cadvisorApi.EventType]bool{
+			cadvisorApi.EventOom: true,
+		},
+		MaxEventsReturned:    1,
+		ContainerName:        "/", // Root container
+		IncludeSubcontainers: false,
+	}
+	events, err := kl.cadvisor.GetPastEvents(request)
+	if err != nil {
+		return nil, timestamp, err
+	}
+	if len(events) == 0 {
+		glog.V(5).Infof("No sys oom events reported by cadvisor")
+		return nil, timestamp, nil
+	}
+	if len(events) > 1 {
```

Comment:
Seems a bit too defensive to have these checks, but I'm not wholly against them.